### DISCUSSION TITLE
Life support?

### DISCUSSION THREAD

**Discussion by airgap** [romkatv/powerlevel10k#2681] [General]
P10k is now on life support? I am sad.

**Comment by runofthemillgeek** [comment]
Came here to look into an issue I had and saw the message. Hope everything's OK, @romkatv.

**Comment by henderea** [comment]
Are there any plans for a future version?  And will pull requests be applied if anyone wants to fix a bug or add a feature?

**Comment by romkatv** [comment]
I have no plans for future versions or for new features. I won't be replying to requests for help, which encompass well over 90% of all issues I receive. I won't be merging PRs except those rare ones that can be merged as-is. I won't be fixing bugs other than in exceptional circumstances.

That said, I am using powerlevel10k myself and intend to continue using it. I won't be breaking powerlevel10k and I don't expect it to break on its own. It's not the kind of software that requires active maintenance just to keep working. If it works now for you, it'll keep working.

**Comment by moritzdietz** [comment]
I'm sorry you had to make this choice and I respect it. Hope you're doing well and if not, hope you will get better.

**Comment by romkatv** [comment]
I appreciate your concerns for my wellbeing. I am well by any objective and subjective measure.

**Comment by grimm26** [comment]
I just want to rip out `gitstatus` and replace with just `vcs_info` and `git status`. I've started looking at it, but I need to figure out how to untangle the web :)

**Comment by romkatv** [comment]
You don't have to "rip out" anything. There is a public API for custom prompt segments. See `p10k help segment`.

**Comment by grimm26** [comment]
so just write a new prompt segment that doesn't use gitstatus. I guess that's easier than converting the existing one :)

**Reply by romkatv** [reply]
Yes, it would be a lot easier for you to write a custom prompt segment than to modify the builtin `vcs` segment.

**Reply by Syphdias** [reply]
I'm curious: Why do you not want to use `vcs`?
If this is purely about the "life support" "issue", I'd encourage you to subscribe to all notifications of this repo. Roman is anything but inactive and helpful where ever he can.

**Reply by grimm26** [reply]
what I want is to excise gitstatus - as does Roman. Roman has indicated that he is pretty much done with all of this, though, so I will forge ahead as you do with an open source project.

**Reply by romkatv** [reply]
I want to have a *fast* backend for `vcs` segment that does not rely on gitstatusd. This will allow me to stop supporting gitstatusd, which is quite costly given that `git` is constantly evolving.

Note that `vcs` already has a *slow* backend based on `vcs_info`. You can enable it with `POWERLEVEL9K_DISABLE_GITSTATUS=true`. I don't think you'll gain anything by doing it but you'll definitely feel the slowdown.

**Reply by grimm26** [reply]
I don't really notice the slowdown but I don't like the default styling/format. I'll see about trying to match what I have with gitstatus. I gain not having to rely on gitstatus and having 10 or 20 of those processes hanging around as I work through the day.

**Reply by romkatv** [reply]
> I gain not having to rely on gitstatus and having 10 or 20 of those processes hanging around as I work through the day.

The alternative to keeping processes hanging around is to fork and terminate them on each command, which is exactly what `vcs_info` does. It's wasteful of resources.

Think of it this way. If you are actively using a shell, gitstatusd saves resources by not having to fork and terminate a bunch of processes on every command. If you aren't actively using a shell, gitstatusd is idle, and thus using virtually no resources.

If your concern is efficiency, replacing gitstatus with `vcs_info` would be counterproductive.

**Comment by jordantrizz** [comment]
Thoughts on a fork? powerlevel11k? I use it day to day, but ultimately I'm not @romkatv and although I can zsh and bash myself, taking something like this on wouldn't be a good idea, nor would I have the time. But if things start breaking, it would become a priority to fix things. The only reason I'm posting this is due to https://github.com/romkatv/powerlevel10k/issues/2513#issuecomment-3193650394

So perhaps the best way forward is pushing PR's and hope for feedback and corrections that can then be merged.

Thanks for all your work @romkatv

**Reply by romkatv** [reply]
See https://github.com/microsoft/vscode/issues/271897#issuecomment-3414463634

**Reply by jordantrizz** [reply]
Thanks @romkatv I see that commit was from 2014? So just updating powerlevel10k is the resolution for people that are running an older version. I was actually trying to figure out how to tell what version I was on, but p10k doesn't reveal this. I can see there is a typeset for the version, which I can just echo.

**Reply by romkatv** [reply]
```text
% print -r -- $P9K_VERSION
1.20.15
```

**Reply by jordantrizz** [reply]
Yea, that's what I did. I was going to submit a PR to add this into the p10k command. I used Github Copilot, don't know if that's acceptable for you, if not all good. I also noticed the version number and tagged releases isn't sync. I understand you're not doing anymore work on this, but if you want help with house keeping just let me know.

**Reply by romkatv** [reply]
Thanks for offering help. It's not needed in this case.

`P9K_VERSION` is just like `ZSH_VERSION`. It's a fine interface.

I cut p10k releases every now and then.

**Reply by jordantrizz** [reply]
The only reason I mentioned it is because running p10k command shows no version, the variable is P9K_VERSION but some might be confused and think its P10K_VERSION instead. Also check the repo and seeing releases, and it not being 1.20.15 might be confusing.

Just food for thought.
